relentless/optimize/objective.py

```diff
@@ -165,3 +181,150 @@ def design_variables(self):
         :class:`ObjectiveFunction` for which the result was constructed, mapped
         to the value of the variables at the time the result was constructed."""
         return self._design_variables
+
+class RelativeEntropy(ObjectiveFunction):
+    r"""Relative entropy.
+
+    The relative entropy quantifies how different two probability distributions are.
+    This can be applied to the probability distributions of a target ensemble
+    and the ensemble of a simulated molecular system, which is parametrized
+    by some pair potential functions.
+
```

Comment:
I think we should add another paragraph here describing how this works. The two key points are:
* The target is known in advance.
* The model ensemble is generated by a simulation for a given set of parameters.